sycl/source/backend.cpp

```diff
@@ -95,6 +95,28 @@ queue make_queue_impl(pi_native_handle NativeHandle, const context &Context,
       std::make_shared<queue_impl>(PiQueue, ContextImpl, Handler));
 }
 
+queue make_queue_impl2(pi_native_handle NativeHandle, const context &Context,
+                       RT::PiDevice Device, bool KeepOwnership,
+                       const property_list &PropList, bool IsImmCmdList,
+                       const async_handler &Handler, backend Backend) {
+  const auto &Plugin = getPlugin(Backend);
+  const auto &ContextImpl = getSyclObjImpl(Context);
+
+  // Create PI properties from SYCL properties.
+  RT::PiQueueProperties Properties[] = {
+      PI_QUEUE_FLAGS,
+      queue_impl::createPiQueueProperties(PropList, QueueOrder::OOO), 0, 0, 0};
```

Comment:
Could you please clarify why we pass `QueueOrder::OOO` here? Shouldn't we infer this from `PropList`?

Reply:
There is an existing instance of this in queue_impl.hpp. I am leveraging existing functionality for creating the PI queue properties. I believe this arises from the fact that there is no SYCL property for out-of-order, only a property for in-order.

Reply:
What if a user passes `sycl::queue::in_order`? Will we still pass `PI_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE` to the low level runtime?

Reply:
This was an error, now corrected. The SYCL runtime was reporting in-order/out-of-order as expected, but within the plugin the queue was always out-of-order. This has been corrected.